src/Core/Extensible.php

```diff
@@ -553,8 +567,12 @@ public function getExtensionInstances()
 
             if ($extensions) {
                 foreach ($extensions as $extension) {
-                    $instance = Injector::inst()->get($extension);
-                    $this->extension_instances[get_class($instance)] = $instance;
+                    list($extensionName) = ClassInfo::parse_class_spec($extension);
```

Comment:
Can you copy the approach used in `Injector::normaliseArguments()` to skip calling `ClassInfo::parse_class_spec()` if the definition doesn’t contain a `(`?

Might be safest to also include the `$name = trim($name);` from that method too, just in case...

Reply:
done